packages/devtools_app/lib/src/memory/flutter/memory_heap_tree_view.dart

```diff
@@ -749,67 +914,119 @@ class MemorySnapshotTableState extends State<MemorySnapshotTable>
       setState(() {});
     });
 
+    addAutoDisposeListener(controller.selectTheSearchNotifier, () {
+      setState(() {
+        _trySelectItem();
+      });
+    });
+
     addAutoDisposeListener(controller.searchNotifier, () {
       setState(() {
```

Comment:
nit: add the setState calls to _trySelectItem when the state changes rather than wrapping both calls to _trySelectItem with setState.

Reply:
I think this is what you want.  If not I think this is better.

```
      if (_trySelectItem()) {
        setState(() {
          closeAutoCompleteOverlay();
        });
      }
```